plugins/scaffolder/src/components/fields/OwnerPicker/OwnerPicker.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useEffect } from 'react';
+import { Field } from '@rjsf/core';
+import { catalogApiRef } from '@backstage/plugin-catalog-react';
+import { useApi, Progress } from '@backstage/core';
+import { useAsync } from 'react-use';
+import Select from '@material-ui/core/Select';
+import InputLabel from '@material-ui/core/InputLabel';
+import FormControl from '@material-ui/core/FormControl';
+import FormHelperText from '@material-ui/core/FormHelperText';
+import { Entity } from '@backstage/catalog-model';
+
+const entityRef = (entity: Entity | undefined): string => {
+  if (!entity) {
+    return '';
+  }
+  const {
+    kind,
+    metadata: { namespace, name },
+  } = entity;
+  return `${kind}:${namespace}/${name}`;
+};
+
+export const OwnerPicker: Field = ({
+  onChange,
+  uiSchema,
+  rawErrors,
+  formData,
+}) => {
+  const allowedKinds = (uiSchema['ui:options']?.allowedKinds || [
+    'User',
+    'Group',
+  ]) as string[];
+  const catalogApi = useApi(catalogApiRef);
+
+  const { value: owners, loading } = useAsync(() =>
+    catalogApi.getEntities({ filter: { kind: allowedKinds } }),
+  );
+
+  const sortedOwners = owners?.items.sort((a, b) => {
+    const aRef = entityRef(a);
+    const bRef = entityRef(b);
+
+    if (aRef < bRef) return -1;
+
+    return aRef === bRef ? 0 : 1;
+  });
+
+  const defaultValue = entityRef(sortedOwners?.[0]);
+
+  useEffect(() => {
+    if (!formData) {
+      onChange(defaultValue);
+    }
+  }, [onChange, formData, defaultValue]);
+
+  if (loading) {
+    return <Progress />;
+  }
+
+  const onSelection = (
+    evt: React.ChangeEvent<{ name?: string; value: unknown }>,
+  ) => onChange(evt.target.value as string);
+
+  return (
+    <>
+      <FormControl
+        margin="normal"
+        required
+        error={rawErrors?.length > 0 && !formData}
+      >
+        <InputLabel htmlFor="hostInput">Owner</InputLabel>
+        <Select
+          native
+          id="hostInput"
+          onChange={onSelection}
+          value={formData || defaultValue}
+        >
+          {sortedOwners ? (
+            sortedOwners.map(entity => {
+              const ref = entityRef(entity);
+              return (
+                <option key={ref} value={ref}>
+                  {entity.kind}: {entity.metadata.name}
```

Comment:
Can this also support namespaces? ie. add a `<namespace>/<name>` prefix if the entity is not in the default namespace? This will differentiate duplicate entity names under different namespaces

Reply:
That's actually a good idea. As someone who doesn't use namespaces at all I didn't want the noise but omitting in the case of `default` is a good compromise.